inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -66,6 +69,210 @@ namespace details {
         return nullptr;\
     });\
 
+/// \brief Creates legacy representation of CNNLayer for SubGraphOp.
+/// \param layer node type
+/// \return pointer to CNNLayer with legacy representation of SubGraphOp.
+CNNLayer::Ptr createSubGraphLayer(const std::shared_ptr<ngraph::Node>& layer) {
+    auto sub_graph = std::dynamic_pointer_cast<ngraph::op::util::SubGraphOp>(layer);
+    if (!sub_graph) {
+        THROW_IE_EXCEPTION << "Cannot cast layer to SubGraphOp.";
+    }
+
+    // inputs/outputs of TensorIterator (ngraph representation)
+    auto parameters = sub_graph->get_function()->get_parameters();
+    auto results = sub_graph->get_function()->get_results();
+
+    // Convert body (ngraph representation) to CNNNetwork.
+    // This network will contain nodes of type = "Input" and data nodes with wrong names.
+    // IE TensorIterator doesn't include such nodes so we create CNNNetwork in a separate scope
+    // to call the destructor and delete these "Input"/data nodes.
+
+    TensorIterator::Body body;
+    {
+        InferenceEngine::CNNNetwork body_net(sub_graph->get_function());
+        InferenceEngine::CNNNetwork net(InferenceEngine::details::convertFunctionToICNNNetwork(body_net.getFunction(), body_net));
+        // Paranoid check for cycles
+        bool res = CNNNetForestDFS(
+            CNNNetGetAllInputLayers(net), [](const CNNLayerPtr& layer) {}, false);
+        if (!res) {
+            THROW_IE_EXCEPTION << "Loop detected. SubGraphOp body should not contain loops.";
+        }
+
+        // Get inputs/outputs of cnn network
```

Comment:
Such comment might suggest that below part of code should be moved to a function.

Reply:
It's just information that variables below are the input/output of cnn network.